# Benchmarking-taxonomic-classifiers-for-soil-shotgun-data

This repository contains the final version of scripts used to prepare data for simulation, simulation scripts, alongside the prepared pipelines and classifier scripts used to process the simulated sequencing data for the publication on Benchmarking taxonomic classifiers for accurate profiling of fungal, bacterial and archaeal communities in soil shotgun metagenomic data using in-silico tools. Many of the scripts used are modified version of scripts previously made by Amy H Fitzpatrick available at https://github.com/ahfitzpa/Benchmarking-bioinformatics-norovirus-amplicons

Pipelines compared include Kaiju, Kraken, Kraken with custom gtdb-tk database, Metaphlan3 and Metaphlan4.

The scripts and tools for fungal, bacterial and archaeal genomes downloaded for the main mock community can be found at https://github.com/roseedwin/Bacteria-Archaea-genomes-download and https://github.com/roseedwin/Fungal-genome-downloads

And custom database used for Kraken is the GTDB-TK database released 2022 April and Fungal genomes belonging to phylum Basidiomycota, Mucoromycota and Ascomycota as available in NCBI assembly database on 20th January 2023
